apps/mobile/src/screens/chat/components/categories-form.tsx

```diff
@@ -0,0 +1,118 @@
+import React from "react";
+
+import { Button, Checkbox, Text, View } from "~/libs/components/components";
+import {
+	useAppDispatch,
+	useAppForm,
+	useAppSelector,
+	useCallback,
+	useEffect,
+	useFormController,
+	useState,
+} from "~/libs/hooks/hooks";
+import {
+	DEFAULT_ERROR_MESSAGE,
+	toastMessage,
+} from "~/libs/packages/toast-message/toast-message";
+import { globalStyles } from "~/libs/styles/styles";
+import { quizCategoriesValidationSchema } from "~/packages/quiz/quiz";
+import { actions as quizActions } from "~/slices/quiz/quiz";
+
+const CategoriesForm: React.FC = () => {
+	const dispatch = useAppDispatch();
+	const { categories } = useAppSelector((state) => state.quiz);
+
+	const ZERO_CATEGORIES = 0;
+	const hasCategories = Boolean(categories.length > ZERO_CATEGORIES);
+
+	useEffect(() => {
+		if (!hasCategories) {
+			void dispatch(quizActions.getQuizCategories());
+		}
+	}, [dispatch, hasCategories]);
+
```

Comment:
Suggestion to simplify and not use hasCategories.
Also this useEffect should not be in the component. It will be in the screen. So it is better to move the component to src/libs/components

Reply:
Okay, I agree it's better to pass `categories` as a prop, but I have doubts about moving `CategoriesForm` to the common components, as it only makes sense if we plan to use this form not just on the chat screen. Maybe it's better to keep `CategoriesForm` next to the parent screen and not overload the common components?

Reply:
Now I see this component on two screens (No smth else seems unnecessary there now). Therefore, I would add this component to the common ones.
<img width="179" alt="Screenshot 2024-09-03 at 11 48 09" src="https://github.com/user-attachments/assets/ce7aec77-dde7-45ba-ba1c-2861c67f38a9"> <img width="179" alt="Screenshot 2024-09-03 at 11 48 32" src="https://github.com/user-attachments/assets/eb75740f-ed11-4d5b-9085-2cb8634c5d2b">

Reply:
In this case, I agree absolutely